frontend/shared/src/hooks/useGoToPage.ts

```diff
@@ -8,11 +8,10 @@ const useGoToPage = (): GoToPageFunction => {
   const router = useRouter();
   const locale = useLocale();
   const isRouting = useIsRouting();
-
   return React.useCallback(
     (pagePath, options): void => {
-      if (isRouting) {
-        // if already routing dont re-route;
+      // if already routing, or we're trying to route to same path, do nothing;
+      if (isRouting || pagePath === router.pathname) {
```

Comment:
What about the case if the query parameters aren't the same?

Reply:
fixed now. Let's us `router.asPath` which checks the query param as well

Reply:
![Screenshot 2023-01-11 at 3 42 06 PM](https://user-images.githubusercontent.com/7648194/211835995-5e9f1a13-86ea-410a-8ce7-80acfe53e1f9.png)

Reply:
That's good